web/src/lib/components/forms/library-scan-settings-form.svelte

```diff
@@ -48,13 +48,16 @@
     }
 
     try {
-      library.exclusionPatterns.push(exclusionPatternToAdd);
+      // Check so that exclusion pattern isn't duplicated
+      if (!library.exclusionPatterns.includes(exclusionPatternToAdd)) {
+        library.exclusionPatterns.push(exclusionPatternToAdd);
+        exclusionPatterns = library.exclusionPatterns;
+      }
```

Comment:
We're doing that check in 3 different places or so. Not sure if I'm a huge fan of this...

Reply:
We check both in the modal for add/edit and in the handle function. I think both are needed. I'll likely also add a check in the server to prevent duplicates submitted via API.

Reply:
Another option is to not do anything client side and auto dedupe server side instead

Reply:
We still need to dedupe client side because of the client-side bugs in #6694. I'll address the server side in another PR